accumulator/pollardutil_test.go

```diff
@@ -0,0 +1,45 @@
+package accumulator
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestPollardSerializeDeserialize(t *testing.T) {
+	var p, q Pollard
+	// generate slice of leaf
+	leaves := make([]Leaf, 10)
+	for i := 0; i < len(leaves); i++ {
+		leaves[i].Hash[0] = uint8(i + 1)
+	}
+	// add leaves to pollard
+	err := p.add(leaves)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// new pollard
+	err = q.add(leaves)
```

Comment:
Why add leaves to q, as you then deserialize into q?
Actually maybe this should lead to an error if you try to deserialize into an already initialized / filled pollard.

Reply:
So  should I first serialize q then perform deserailization and then compare .

Reply:
The test should check if the serialized bytes indeed deserialize correctly from the bytes. So there should be no `q.add()` since this will negate the deserialization test.

If we do the `q.add()`, the test will pass even if `err = q.Deserialize(old_byte)` fails.